onnxruntime/core/xnnpack/optimizer/xnnpack_transformer.cc

```diff
@@ -0,0 +1,497 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+#include "core/xnnpack/optimizer/xnnpack_transformer.h"
+
+#include <deque>
+
+#include "core/framework/op_node_proto_helper.h"
+#include "core/graph/graph_utils.h"
+#include "core/graph/node_attr_utils.h"
+#include "core/graph/schema_registry.h"
+#include "core/optimizer/initializer.h"
+#include "core/optimizer/nhwc_transformer.h"
+#include "core/optimizer/selectors_actions/helpers.h"
+#include "core/optimizer/transpose_optimizer/optimizer_utils.h"
+#include "core/optimizer/utils.h"
+#include "core/providers/cpu/nn/pool_attributes.h"
+
+using namespace ONNX_NAMESPACE;
+using namespace ::onnxruntime::common;
+using namespace onnx_layout_transformation;
+
+#define XNNPACK_RETURN_IF_ERROR(expr)                                                                                 \
+  do {                                                                                                                \
+    auto _status = (expr);                                                                                            \
+    if ((!_status.IsOK())) {                                                                                          \
+      std::ostringstream oss;                                                                                         \
+      oss << __FILE__ << ":" << __LINE__ << " Convert the model to XNNPack graph failed: " << _status.ErrorMessage(); \
+      return Status(_status.Category(), _status.Code(), oss.str());                                                   \
+    }                                                                                                                 \
+  } while (0)
+
+namespace onnxruntime {
+static bool IsPaddingTypeSupported(AutoPadType auto_pad) {
+  return auto_pad == AutoPadType::NOTSET || auto_pad == AutoPadType::VALID || auto_pad == AutoPadType::SAME_UPPER;
```

Comment:
actually, we can support SAME_LOWER by setting pads explicitly.

Reply:
I will consider it later. Models converted from TF do not have SAME_LOWER .  I searched around it is even hard to find such a model. I guess pytorch has it.